operations/src/main/java/com/testingblaze/http/RestfulWebServices.java

```diff
@@ -255,4 +257,89 @@ private void consoleLogger(CallTypes callType, Response response) {
+     * @param dontCheckTheseKeys add Set of keywords that should not be part of validation...
+     *                           e.g
+     *                           {
+     *                              "name":{
+     *                                  "firstname":"jitu",
+     *                                  "lastname":"pisal
+     *                              }
+     *                           }
+     *   if we don't want to validate "lastname" then add it in the Set<String>
+     */
+    private void validateJsonResponse(JsonObject expectedResponse, JsonObject actualResponse, Set<String> dontCheckTheseKeys) {
+        for (String key : actualResponse.keySet()) {
+            if (actualResponse.get(key) instanceof JsonObject) {
+                JsonObject expectedChildResponseObject = expectedResponse.getAsJsonObject(key);
+                JsonObject childResponseObject = actualResponse.getAsJsonObject(key);
+                validateJsonResponse(expectedChildResponseObject, childResponseObject, dontCheckTheseKeys);
+            } else if (actualResponse.get(key) instanceof JsonArray) {
+                JsonArray childResponseArray = actualResponse.getAsJsonArray(key);
+                JsonArray expectedChildResponseArray = expectedResponse.getAsJsonArray(key);
+                for (int i = 0; i < childResponseArray.size(); i++) {
+                    if (childResponseArray.get(i) instanceof JsonObject) {
+                        validateJsonResponse((JsonObject) expectedChildResponseArray.get(i), (JsonObject) childResponseArray.get(i), dontCheckTheseKeys);
+                    } else {
+                        if (!dontCheckTheseKeys.contains(key)) {
+                            if (actualResponse.get(String.valueOf(childResponseArray.get(i))) == null) {
+                                Assert.assertEqualsNoOrder(new Gson().fromJson(actualResponse.get(key), String[].class),
+                                        new Gson().fromJson(expectedResponse.get(key), String[].class));
+                            } else {
+                                Assert.assertEquals(childResponseArray.get(i),
+                                        expectedChildResponseArray.get(i),
+                                        "Failed while validating the response" +
+                                                actualResponse.get(key.toString()));
+                            }
+                        }
+                    }
+                }
+            } else if (!dontCheckTheseKeys.contains(key)) {
+                Assert.assertEquals(actualResponse.get(key).toString(), expectedResponse.get(key).toString(),
+                        "Failed while validating the response" + actualResponse.get(key.toString()));
+            }
+        }
+    }
+
+    /**
+     * Converts String json response to JsonObject(gson)
+     * @param string json response
+     * @return JsonObject
+     */
+    private static JsonObject getJSONObject(String string) {
```

Comment:
Do we allready have gsonToJsonConvertoor method in this class? If it exissts, i think that what it does. Let me know on it.

Reply:
Please respond.

Reply:
We don't have any method to convert json string to jsonObject.
Removed this method from here to FileHandler.java.